### PR TITLE
Add contributing doc

### LINKED ISSUE #12: Repo structure (contribution)
For people that want to contribute, we should document the different repositories and how they map to the overall architecture.

Example:

- **REST API**
    - `api` - the primary API for Screwdriver
    - `plugin-*` - HAPI plugins that validate/enforce input/output as well as authentication/authorization
    - `models` - Business logic between the Datastore and Execution Layer
    - `data-schema` - Definitions of the internal data structure

- **Web UI**
    - `ui`

- **Launcher**
    - `job-tools` - docker container with the self-contained screwdriver launcher
    - `launcher-execute` - cli to execute `screwdriver.yaml` steps

- **Execution Engine**
    - `executor-*`

- **Datastore**
    - `datastore-*`

- **Shared**
    - `config-parser` - Tool to parse the `screwdriver.yaml`

### PR DESCRIPTION
**Pull request by petey**
Add contributing doc, along with descriptions of what each repo is for. Fixes #12

### CODE REVIEW

**Comment by stjohnjohnson** [comment]
👍 
worstfirstpr

File: docs/community/contributing.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+* **[datastore-imdb][datastore-imdb-repo]**: In-memory Implementation [![Version][datastore-imdb-npm-image]][datastore-imdb-npm-url] [![Open Issues][datastore-imdb-issues-image]][datastore-imdb-issues-url]
+
+
+### [Config Parser][config-parser-repo] [![Version][config-parser-npm-image]][config-parser-npm-url] [![Open Issues][config-parser-issues-image]][config-parser-issues-url]
+
+Node module for validating and parsing user's `screwdriver.yaml` configurations.
+
+### [Guide][guide-repo] [![Open Issues][guide-issues-image]][guide-issues-url]
+
+This documentation! Everything you ever hoped to know about the screwdriver project.
```
Comment (tkyi) [inline docs/community/contributing.md]: capitalize `Screwdriver`

**Comment by FenrirUnbound** [comment]
👍 

as2bsqa

**Comment by d2lam** [comment]
👍

**Comment by tkyi** [comment]
Just wondering if we should mention the dynamic-dynamodb repo.

Otherwise, LGTM! 👍 
giphy-3

**Pull request merged**